### PR TITLE
add cubemap support

### PR DESCRIPTION
**Pull request by JerryChCN**
it implements Cubemap binding with texture3d fixed issue #644. it's the easiest way if don't refactor the texture. the project doesn't have an example showing how to test it correctly, so it may have some problems or bugs.

### CODE REVIEW

File: src/texture/mod.rs [hunk 1]
```diff
@@ -789,6 +789,75 @@ impl<P> Texture3dDataSink<P> for Vec<Vec<Vec<P>>> where P: Copy + Clone {
...
+        pos_z: RawImage2d<'a, T>,
+        neg_z: RawImage2d<'a, T>,
+    ) -> Self {
+        let mut arr = Vec::<RawImage2d<'a, T>>::with_capacity(6);
+        let dimension = pos_x.width;
+        arr.push(pos_x);arr.push(neg_x);
+        arr.push(pos_y);arr.push(neg_y);
+        arr.push(pos_z);arr.push(neg_z);
+        arr.iter().for_each(|i| {
+            if i.width != i.height {
```
Comment (est31) [inline src/texture/mod.rs:825]: Can this be an assert_eq instead? assert_eq can emit a format string too.

File: src/texture/mod.rs [hunk 2]
```diff
@@ -789,6 +789,75 @@ impl<P> Texture3dDataSink<P> for Vec<Vec<Vec<P>>> where P: Copy + Clone {
...
+/// wrap RawCubemap in RawImage3d
+pub struct RawCubemap<'a, T: Clone> {
+    data: RawImage3d<'a, T>,
+    dimension: u32,
+}
+
+impl<'a, T: Clone + 'a> RawCubemap<'a, T> {
+    /// transform RawImage3d into RawImage directly
+    pub fn from_texture3d(data: RawImage3d<'a, T>) -> Self {
+        if data.width != data.height { panic!("width does't equal height"); }
```
Comment (est31) [inline src/texture/mod.rs:802]: assert_eq

File: src/texture/mod.rs [hunk 3]
```diff
@@ -789,6 +789,75 @@ impl<P> Texture3dDataSink<P> for Vec<Vec<Vec<P>>> where P: Copy + Clone {
     }
 }
 
+
+/// wrap RawCubemap in RawImage3d
```
Comment (est31) [inline src/texture/mod.rs:793]:
```suggestion
/// Wraps a RawImage3d into a cube map
```

File: src/texture/any.rs [hunk 4]
```diff
@@ -275,7 +290,23 @@ pub fn new_texture<'a, F: ?Sized, P>(facade: &F, format: TextureFormatRequest,
                 a => a
             };
 
-            if storage_internal_format.is_some() && (ctxt.version >= &Version(Api::Gl, 4, 2) || ctxt.extensions.gl_arb_texture_storage) {
+            if bind_point == gl::TEXTURE_CUBE_MAP {
+                let mut cube_data_raw = if let Some((_, ref data)) = data {
```
Comment (est31) [inline src/texture/any.rs:294]: Can't you just use an unwrap here? Maybe `.ref().unwrap()`?

**Review by est31 — CHANGES_REQUESTED** [review]

**Comment by est31** [comment]
@JerryChCN hi have you seen my review comments? If you want this pull request to be merged, please address them.

**Comment by panda2134** [comment]
Hi! Is there any chance for this PR to be merged?

**Comment by est31** [comment]
Closing because no response. If the comments are addressed, I would love to see another PR!

**Pull request closed without merging**